File: string/destination-city/destination-city.go
package destinationcity

func destCity(paths [][]string) string {
	cities := map[string]int{}
	for _, path := range paths {
		cities[path[0]]++
		cities[path[1]]--
	}
	for city, value := range cities {
		if value == -1 {
			return city
		}
	}

	return ""
}

File: string/destination-city/destination-city.ts
function destCity(paths: string[][]): string {
    const cities: { [key: string]: number } = {};

    for (let path of paths) {
        cities[path[0]] = (cities[path[0]] || 0) + 1;
        cities[path[1]] = (cities[path[1]] || 0) - 1;
    }

    for (let city in cities) {
        if (cities[city] === -1) {
            return city
        }
    }

    return ""
};
